erts/emulator/test/num_bif_SUITE.erl

```diff
@@ -778,3 +778,17 @@ id(X) -> X.
 %% Use the printing library to convert to list.
 int2list(Int, Base) when is_integer(Base), 2 =< Base, Base =< 36 ->
     lists:flatten(io_lib:format("~."++integer_to_list(Base)++"B",[Int])).
+
+bin_to_int(Bin) ->
+    Unaligned = erts_debug:unaligned_bitstring(Bin, 3),
+    try binary_to_integer(Bin) of
+        Int ->
+            Int = binary_to_integer(Unaligned),
+            Int
+    catch
+        C:E ->
+            binary_to_integer(Unaligned)
+    end.
```

Comment:
To be really thorough, should we catch the unaligned call as well and verify that we get the same exception.

Reply:
Yes, good idea.